tests/integration/test_postgresql_replica_database_engine_2/test.py

```diff
@@ -1141,6 +1141,97 @@ def test_dependent_loading(started_cluster):
     instance.query(f"DROP TABLE {table} SYNC")
 
 
+def test_partial_table(started_cluster):
+    table = "test_partial_table"
+
+    pg_manager.create_postgres_table(
+        table,
+        "",
+        f"""CREATE TABLE {table} (
+             key integer PRIMARY KEY,
+             x integer DEFAULT 0,
+             y integer,
+             z text DEFAULT 'z');
+         """,
+    )
+    pg_manager.execute(f"insert into {table} (key, x, z) values (1,1,'a');")
+    pg_manager.execute(f"insert into {table} (key, x, z) values (2,2,'b');")
+
+    pg_manager.create_materialized_db(
+        ip=started_cluster.postgres_ip,
+        port=started_cluster.postgres_port,
+        settings=[
+            f"materialized_postgresql_tables_list = '{table}(z, key)'",
+            "materialized_postgresql_backoff_min_ms = 100",
+            "materialized_postgresql_backoff_max_ms = 100",
+        ],
+    )
+    check_tables_are_synchronized(
+        instance, table, postgres_database=pg_manager.get_default_database(), columns=['key', 'z']
+    )
+
+    pg_manager.execute(f"insert into {table} (key, x, z) values (3,3,'c');")
+    pg_manager.execute(f"insert into {table} (key, x, z) values (4,4,'d');")
+
+    check_tables_are_synchronized(
+        instance, table, postgres_database=pg_manager.get_default_database(), columns=['key', 'z']
+    )
+
+def test_partial_and_full_table(started_cluster):
+    table = "test_partial_and_full_table"
+
+    pg_manager.create_postgres_table(
+        table,
+        "",
+        f"""CREATE TABLE {table}1 (
+             key integer PRIMARY KEY,
+             x integer DEFAULT 0,
+             y integer,
+             z text DEFAULT 'z');
+         """,
+    )
+    pg_manager.execute(f"insert into {table}1 (key, x, y, z) values (1,1,1,'1');")
+    pg_manager.execute(f"insert into {table}1 (key, x, y, z) values (2,2,2,'2');")
+    pg_manager.create_postgres_table(
+        table,
+        "",
+        f"""CREATE TABLE {table}2 (
+             key integer PRIMARY KEY,
+             x integer DEFAULT 0,
+             y integer,
+             z text DEFAULT 'z');
+         """,
+    )
+    pg_manager.execute(f"insert into {table}2 (key, x, y, z) values (3,3,3,'3');")
+    pg_manager.execute(f"insert into {table}2 (key, x, y, z) values (4,4,4,'4');")
+
+    pg_manager.create_materialized_db(
+        ip=started_cluster.postgres_ip,
+        port=started_cluster.postgres_port,
+        settings=[
+            f"materialized_postgresql_tables_list = '{table}1(key, x, z), {table}2'",
+            "materialized_postgresql_backoff_min_ms = 100",
+            "materialized_postgresql_backoff_max_ms = 100",
+        ],
```

Comment:
Let's add a query to check a list of columns of result table in materialized postgres database, `select name, type from system.columns where table = {table}1` and same for `{table}2` and make sure that it is indeed not a full set of columns

Reply:
`check_tables_are_synchronized` already ensures that tables have rigth set of columns. I checked this localy

Reply:
check_tables_are_synchronized ensures that certain set of columns exist in a table, but I meant here to check that no other columns apart from the expected ones are there

Reply:
ah, there is `select *` for clickhouse table, so it does also check that